### ISSUE TITLE
Feature Request: VSCode Editor

### ISSUE THREAD

**Issue by AYapejian** [standardnotes/forum#469]
Just came across [stackblitz](https://github.com/stackblitz/core) [(demo)](https://stackblitz.com/edit/sdk-create-project) [(docs)](https://stackblitz.com/docs#generate-and-embed-new-projects) and it would be great to have a stripped down version (probably not multi-file, don't allow extensions maybe? ) of the demo linked as an editor in standardnotes.  Seems like it could serve as a replacement for several different existing editors.

**Comment by mobitar** [comment]
Doesn't look like Stackblitz is open source, but I agree a VSCode editor would be really cool. VSCode is based on [Monaco](https://github.com/Microsoft/monaco-editor) I believe.

**Comment by AYapejian** [comment]
Good catch, for reference [this](https://github.com/stackblitz/core/issues/418) is what I found regarding licensing.  Looks like they have plans to open source but unsure on the particular license to go with.  Hopefully they sort that soon.  Monaco would be cool too but I haven't had a chance to dig into the monaco feature set yet to see how it compares to the stuff stackblitz added.

**Comment by AYapejian** [comment]
After looking into monaco a bit more I agree, between the specific licensing and seemingly wide support for different use cases and options this would be a great base for a new version of a vscode like editor.  I might give a go at throwing together a POC extension although not sure when I'd have time to do that.  Also for those interested their [playground](https://microsoft.github.io/monaco-editor/playground.html#interacting-with-the-editor-adding-an-action-to-an-editor-instance) is a good resource to get some ideas

**Comment by AYapejian** [comment]
Here is another project I came across that could serve as either inspiration or a base, MIT license and easily run with `docker run -t -p 127.0.0.1:8443:8443 -v "${PWD}:/root/project" codercom/code-server code-server --allow-http --no-auth` [code-server](https://github.com/codercom/code-server)

**Comment by mobitar** [comment]
Very cool!

**Comment by beeman** [comment]
Another open source project that's worth checking out is [Theia](https://theia-ide.org)!
